webapp/components/menu-component/menu-component.vue

```diff
@@ -0,0 +1,42 @@
+<template>
+  <div class="menuButton">
+    <template v-if="loggedIn">
+      <button @click="logoutUser" id="logoutButton">Logout</button>
+    </template>
+    <template v-else>
+      <NuxtLink to="/login" id="loginButton">Login</NuxtLink>
+    </template>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions } from "vuex"
+
+export default {
+  name: "MenuComponent",
+  computed: {
+    ...mapGetters('auth', ['loggedIn']),
+  },
+  methods: {
+    ...mapActions('auth', ['logout']),
+    ...mapActions('auth', ['cookieLogin']),
+    logoutUser() {
+      this.logout()
+      this.$apolloHelpers.onLogout()
+    }
+  },
+  created () {
+    //TODO: make this work in tests
+    //let token = this.$apolloHelpers.getToken()
+    //if (token) this.cookieLogin({token})
```

Comment:
You app has this slight glitch that a page reload shows the unauthenticated, server-side rendered HTML for a second. Then the client-side HTML takes over and I see the authenticated page.

If outlined a solution in the exercise description, how to fix that:
```js
// in store/index.js
import cookie from 'cookie'

export const actions = {
  nuxtServerInit(store, context) {
    const { req } = context.ssrContext
    if (!req) return // static site generation
    const parsedCookies = cookie.parse(req.headers.cookie)
    const token = parsedCookies['apollo-token']
    if (!token) return
    store.commit('auth/setToken', token)
  },
}
```
See here: https://github.com/Systems-Development-and-Frameworks/homework/tree/main/exercises/7#instructions

Learn about `nuxtServerInit`: https://nuxtjs.org/docs/2.x/directory-structure/store/#the-nuxtserverinit-action